decoy/verifier.py

```diff
@@ -30,8 +30,9 @@ def verify(
 
         for i in range(len(calls)):
             calls_subset = calls[i : i + len(rehearsals)]
+            matches = [match_call(c, r) for c, r in zip(calls_subset, rehearsals)]
 
-            if calls_subset == rehearsals:
+            if len(calls_subset) == len(rehearsals) and all(matches):
```

Comment:
Switch condition order for clarity

```suggestion
            if all(matches) and len(calls_subset) == len(rehearsals):
```